module/apmawssdkgo/session.go

```diff
@@ -67,17 +77,35 @@ func send(req *request.Request) {
 		return
 	}
 
-	ctx := req.Context()
+	var (
+		err error
+		ctx = req.Context()
+	)
 
 	tx := apm.TransactionFromContext(ctx)
 	if tx == nil {
 		return
 	}
 
-	bucketName := getBucketName(req)
+	var svc service
+	switch spanSubtype {
+	case serviceS3:
+		svc = newS3(req)
+	case serviceDynamoDB:
+		svc, err = newDynamoDB(req)
+		if err != nil {
+			// The only error case is if TableName is not present
+			// on the struct, but it's a required value.
+			// TODO: How do we handle an error? Or is this being
+			// overly defensive?
```

Comment:
I'd be inclined to create a span with the name `DynamoDB $Operation` if there's no table name. Not sure what other agents are doing. It's probably also fine to not create a span for now.

Reply:
As far as I can see it's always required, I figure we create the span and if there's an edge case somewhere, an enterprising user will report it

Reply:
SGTM